src/App.tsx

```diff
@@ -1,14 +1,48 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { debounce } from 'lodash';
+// import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './components/DropdownMenu/DropdownMenu';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [isListVisible, setIsListVisible] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedUser, setSelectedUser] = useState<Person | null>(null);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery]);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
```

Comment:
Also you could create a hook-file useDebounce and call useCallback there. In this case you would just call ur new hook and pass function there. It would look like this:
```js
const applyQuery= useDebounce(setAppliedQuery, 1000);
```
Here you can find more about useDebounce
https://www.developerway.com/posts/debouncing-in-react